islandora_solr_config/includes/atom_results.inc

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/**
+ * @file
+ * Contains methods to search solr and display results.  depends on
+ * Apache_Solr_Php client.
+ */
+
+/**
+ * Extention of IslandoraSolrResults for templating purposes.
+ * This overrides the displayResults function to provide an alternate display
+ * type.
+ */
+
+class IslandoraSolrResultsAtom extends IslandoraSolrResults {
+
+  /**
+   * Outputs results basically in the normal way.
+   *
+   * Thumbnails pulled from the Fedora repository.
+   *
+   * @param object $islandora_solr_query
+   *   A solr query object.
+   *
+   * @return string
+   *   html output for the resultset. Note: we currently create this
+   *   output manually, should refactor to use drupal forms api.
+   */
+  public function printAtom($islandora_solr_query, $title = "Search Results") {
+
+    global $base_url;
+
+    drupal_add_http_header('Content-Type', 'application/atom+xml; charset=utf-8');
+    $islandora_solr_query->solrLimit = variable_get('islandora_solr_config_rss_limit', 50);
+    $islandora_solr_query->executeQuery();
+
+    // Get raw results.
+    $solr_result = $islandora_solr_query->islandoraSolrResult;
+
+    // All results.
+    $docs = $solr_result['response']['objects'];
+
+    // Loop over results.
+    $items = NULL;
+    foreach ($docs as $doc) {
+      // Turn arrays into strings.
+      foreach ($doc['solr_doc'] as $key => $value) {
+        if (is_array($value)) {
+          // Turn array into comma separated string and trim.
+          $doc[$key] = trim(implode(', ', $value));
+        }
+        else {
+          // Give it a trim.
+          $doc[$key] = trim($value);
+        }
+      }
+      // Get the variables for the <item> element.
+      $item = $this->rssItem($doc);
```

Comment:
This method probably shouldn't be called `rssItem` since it's for atom, maybe something like `atomItem`.